middleware/src/system/system.go

```diff
@@ -0,0 +1,53 @@
+//package systemenv provides functionality to get data such as open ports and services running from the system the middleware is deployed on
+package environment
+
+// SystemEnv provides some information on the system we are running on.
+type SystemEnv struct {
+	network                                            string
+	electrsRPCPort                                     string
+	bitcoinRPCUser, bitcoinRPCPassword, bitcoinRPCPort string
+	lightningRPCPath                                   string
+}
+
+// NewSystemEnv returns a new SystemEnv instance.
+func NewSystemEnv(bitcoinRPCUser, bitcoinRPCPassword, bitcoinRPCPort, lightningRPCPath, electrsRPCPort, network string) SystemEnv {
```

Comment:
This initializing function just sets the value of the struct and has no further logic, so I'd suggest removing it and just letting the callers set the fields of the struct as appropriate from the nil value.